### PR TITLE
Add password possibility to await_install.pm for s390x

### PR DESCRIPTION
**Pull request by skriesch**
- Related ticket: https://progress.opensuse.org/issues/93949#change-418507
- Needle: https://github.com/os-autoinst/os-autoinst-needles-opensuse/blob/21fb2464cd0030c7ba1aabafa8594b39ed585366/await_install-gui-installation-20210712.png
- Verification run(s): 
- https://openqa.opensuse.org/tests/1837750
- https://openqa.opensuse.org/tests/1837830
- https://openqa.opensuse.org/tests/1837838

### CODE REVIEW

**Comment by Vogtinator** [comment]
`/mnt/` is only available after the installation is past the partitioning, which means that if that didn't happen yet, the configuration won't be written into the system correctly -> race condition. I don't really have an idea how to handle this better from openQA though. Ideally this gets implemented properly in YaST.

**Comment by skriesch** [comment]
> Ideally this gets implemented properly in YaST.

There is a SLE feature request at YaST. But I can not look into that about the status.

**Comment by Vogtinator** [comment]
> We need working tests after 1 month without any Tumbleweed releases for s390x.

Agreed, which is why I'd also be fine with a workaround. It could be a bit more specific and robust though, for instance by checking for a needle which indicates that package installation is ongoing first.

Logging in as `bernhard` over ssh and using `su` to create the file and restart `sshd` would be preferable, maybe it's possible to do that in `reconnect_mgmt_consoles` without changes in `os-autoinst` code?

**Comment by skriesch** [comment]
That's exactly what I tried before and didn't work. That was the reason for treating other methods.

**Comment by Vogtinator** [comment]
> That's exactly what I tried before and didn't work (as expected with my openQA knowledge). That was the reason for trying other methods.

I made a PoC in a branch here: https://github.com/Vogtinator/os-autoinst-distri-opensuse/tree/s390xtest (VR: https://openqa.opensuse.org/tests/1838562)

It's still not great though...

File: tests/installation/await_install.pm [hunk 1]
```diff
@@ -125,6 +125,13 @@ sub run {
     }
     # on s390 we might need to install additional packages depending on the installation method
     if (check_var('ARCH', 's390x')) {
+      #For poo#93949, we need password possibility
+      if (!is_sle && !is_leap) {
```
Comment (Vogtinator) [inline tests/installation/await_install.pm:129]: Not imported FWICT

Comment (skriesch) [reply tests/installation/await_install.pm:129]: What do you mean with "not imported"? That is a test of whether Tumbleweed is used.

Comment (skriesch) [reply tests/installation/await_install.pm:129]: Leap and SLE are using the old method.

Comment (Vogtinator) [reply tests/installation/await_install.pm:129]: Needs `use version_utils qw(:VERSION :BACKEND is_sle is_leap);` at the top

Comment (skriesch) [reply tests/installation/await_install.pm:129]: Done

File: tests/installation/await_install.pm [hunk 2]
```diff
@@ -125,6 +125,13 @@ sub run {
     }
     # on s390 we might need to install additional packages depending on the installation method
     if (check_var('ARCH', 's390x')) {
+      #For poo#93949, we need password possibility
```
Comment (Vogtinator) [inline tests/installation/await_install.pm:128]:
```suggestion
      # For poo#93949, we need password possibility
```

File: tests/installation/await_install.pm [hunk 3]
```diff
@@ -125,6 +125,13 @@ sub run {
...
+      if (!is_sle && !is_leap) {
+        select_console 'install-shell';
+        assert_script_run("mkdir -p /mnt/etc/ssh/sshd_config.d");
```
Comment (Vogtinator) [inline tests/installation/await_install.pm:131]: To fail early if the race is lost:

```suggestion
        assert_script_run('mountpoint /mnt && mkdir -p /mnt/etc/ssh/sshd_config.d');
```

File: tests/installation/await_install.pm [hunk 4]
```diff
@@ -125,6 +125,13 @@ sub run {
     }
     # on s390 we might need to install additional packages depending on the installation method
     if (check_var('ARCH', 's390x')) {
+      # For poo#93949, we need password possibility
+      if (!is_sle && !is_leap) {
+        select_console 'install-shell';
+        assert_script_run("mkdir -p /mnt/etc/ssh/sshd_config.d");
+        assert_script_run("(echo 'PermitRootLogin yes'; echo 'PasswordAuthentication yes'; echo 'ChallengeResponseAuthentication yes') > /mnt/etc/ssh/sshd_config.d/allow-root-with-password.conf");
```
Comment (Vogtinator) [inline tests/installation/await_install.pm:132]: Should be enough:

```suggestion
        assert_script_run('echo PermitRootLogin yes > /mnt/etc/ssh/sshd_config.d/allow-root-with-password.conf');
```

**Comment by skriesch** [comment]
I will combine the commits again.

File: tests/installation/await_install.pm (hunk 4 above)
Comment (skriesch) [reply tests/installation/await_install.pm:132]: Really? Then the default path (created by SUSE QA) should have worked, too. That was not the case 1 month ago. I test it again....

**Comment by skriesch** [comment]
Verification run has been working equal as before.

**Comment by skriesch** [comment]
All commits are combined (squash) now.

File: tests/installation/await_install.pm [hunk 5]
```diff
@@ -125,6 +125,13 @@ sub run {
     }
     # on s390 we might need to install additional packages depending on the installation method
     if (check_var('ARCH', 's390x')) {
+      # For poo#93949, we need password possibility
+      if (!is_sle && !is_leap) {
```
Comment (jknphy) [inline tests/installation/await_install.pm:129]: what about `is_tumbleweed` ?

File: tests/installation/logs_from_installation_system.pm [hunk 6]
```diff
@@ -41,6 +41,7 @@ sub run {
             die "IPL device was not set correctly";
         }
     }
+
```
Comment (jknphy) [inline tests/installation/logs_from_installation_system.pm:44]: probably unintended line break or we need to change the copyright date.

File: tests/installation/await_install.pm (hunk 5 above)
Comment (skriesch) [reply tests/installation/await_install.pm:129]: This if case is Tumbleweed and all what is based on new versions.

Comment (Vogtinator) [reply tests/installation/await_install.pm:129]: I think with `is_sle_micro` this is actually no longer identical to `is_tumbleweed`...

Comment (skriesch) [reply tests/installation/await_install.pm:129]: So I should add _!is_sle_micro_?

File: tests/installation/logs_from_installation_system.pm (hunk 6 above)
Comment (skriesch) [reply tests/installation/logs_from_installation_system.pm]: I tested the code there in the past. I have reverted it now.

Comment (skriesch) [reply tests/installation/logs_from_installation_system.pm]: git checkout HEAD^1 logs_from_installation_system.pm

File: tests/installation/await_install.pm (hunk 5 above)
Comment (skriesch) [reply tests/installation/await_install.pm]: That is added

File: tests/installation/await_install.pm [hunk 7]
```diff
@@ -125,6 +125,13 @@ sub run {
     }
     # on s390 we might need to install additional packages depending on the installation method
     if (check_var('ARCH', 's390x')) {
+      # For poo#93949, we need password possibility
+      if (!is_sle && !is_leap & !is_sle_micro) {
+        select_console 'install-shell';
+        assert_script_run('mountpoint /mnt && mkdir -p /mnt/etc/ssh/sshd_config.d');
+        assert_script_run('echo PermitRootLogin yes > /mnt/etc/ssh/sshd_config.d/allow-root-with-password.conf');
+        select_console 'installation';
+      }
```
Comment (okurz) [inline tests/installation/await_install.pm:134]: please extract this into a function. The run method is already too big and complicated

**Review by okurz — CHANGES_REQUESTED** [review]
The commit message could have a bit more details, e.g. what issue is fixed, why this needs to change, to which versions of which product it applies, link to a bug or a fate entry, the choices you made, etc. Also see https://commit.style/ or http://chris.beams.io/posts/git-commit/ as a helpful guide how to write good commit messages.

What you put into the description of the pull request could also be in the commit message details. I don't know how you created your pull request but your git commit messages has *only* the subject line while the pull request description has more details. Please keep in mind that the github pull request description is only visible on github, the commit can be considered permanent information storage.

I can recommend the tool [hub](https://github.com/github/hub) (zypper in rubygem-hub) for easier PR creation. Also, I myself use a script [git-pr-last](http://github.com/okurz/scripts/tree/master/git-pr-last) to create a PR with proper description for these simple one-commit PRs:
```bash
$ cat $(which git-pr-last )
#!/bin/sh -e
target="${target:-"$USER"}"
git push $target && git show --no-patch --format=%B | hub pull-request -F -
```

See https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/CONTRIBUTING.md#coding-style for more details

File: tests/installation/await_install.pm (hunk 7 above)
Comment (skriesch) [reply tests/installation/await_install.pm]: Extracted into a function. Test verification is running at the moment...

Comment (skriesch) [reply tests/installation/await_install.pm]: Tests are passed again. :)

Comment (okurz) [reply tests/installation/await_install.pm]: ok, this part looks good

**Comment by okurz** [comment]
@skriesch please provide a bit more details in the git commit message as described. You can take a look into existing gits in this repo for examples.

**Comment by skriesch** [comment]
> @skriesch please provide a bit more details in the git commit message as described. You can take a look into existing gits in this repo for examples.

I have learned, that commit messages should not be too long. I have added a body now.

**Review by okurz — APPROVED** [review]

**Comment by okurz** [comment]
Please squash code style changes into the original commit

**Comment by skriesch** [comment]
> Please squash code style changes into the original commit

Done. I wanted to check the test first.

**Pull request merged**